mac/Keyman4MacIM/Keyman4MacIM/KMSettingsRepository.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * KMSettingsRepository.h
+ * Keyman
+ *
+ * Created by Shawn Schantz on 2024-07-29.
+ *
+ */
```

Comment:
Ugh, I thought I was following a pattern that I copied from the standards a while back.  I thought we even had a discussion about the date format. 

I think that author and date can be very helpful to understand the context of the code at a glance without jumping into git. We can talk separately about this, but Is there a strong reason to exclude those?

Reply:
> I think that author and date can be very helpful to understand the context of the code at a glance without jumping into git. We can talk separately about this, but Is there a strong reason to exclude those?

The primary reasons _not_ to do it are:

1. avoid multiple "sources of truth" -- the initial creator of the file may well not be the most relevant, nor the create date
2. avoid busy work

I think we can be pretty confident that git will get it right. In most good code editors, we can see git blame line-by-line, which nicely eliminates the burden of seeing the git data. I would encourage going to git for this kind of metadata -- commit history is really helpful (probably more helpful than original author + date IMO).

The only real downside I can see is if you want to see the original commit: if the header changes over time, then the git blame will show most recent commit against it by default, not the original, and it takes a little more digging to get back to the first commit.

Reply:
I don't think any of these things override the practical value of this tiny piece of metadata: who created this thing and when. It is written once in the life of the file, and only ever updated with maybe a second author if there is a major redesign within the same code. 

In most cases, it will remain as one name and one date but can reveal a lot on its own without scanning the commit history. If it doesn't seem helpful, it can easily be ignored along with the copyright statement. In my experience, the initial author is heavily involved in the design of the code (often the sole person responsible), and a lot of the original design persists even after many edits and attempts to refactor away from the original design.

For the Mac code at least, it looks like the first commit was not the author at all but just the guy who first made use of git. So the real author would be anonymous if he hadn't included his name in the header.

In short, I don't see this as redundant information but something with long-term practical value and maintenance cost that is approximately zero.

The file name and project name does seem like redundant busywork in the header, so I agree that we should remove those. (I've changed the class name many times and forgotten to update the header.)

Reply:
OK I am not overly bothered either way. Personally I feel like it is busy work which replicates what we _now_ have in git. Yes, back in the day we didn't use git but that's a slippery argument because then we end up putting all commit history back into source files because what if we change vcs in the future and cannot transfer commit history and well yuck.

See for example https://github.com/keymanapp/keyman/blob/d8986be12fe0fb2078b93fa4650e2a7fe06f16d1/windows/src/engine/keyman32/keyman32.cpp for what things looked like before we relied on git history. If you scan through the file you'll also see `// I1234` comments which were how we tagged changes. It worked but it was increasingly noisy so it was really good to move that change history into vcs metadata.

tl;dr: If you want to keep it in, OK, and then please also update the code style wiki to say that we can optionally include the `Created by <author> on yyyy-mm-dd` line.

Reply:
Thanks, I find that one extra line to be helpful. I've updated the wiki.